public/script.js

```diff
@@ -1,23 +1,71 @@
-// Find the button
-// when you click the button, it should call a function "getSnippet"
+// all elements for the script from the index .
+const inputs = document.getElementById('paragraphInput');
 const paragraph = document.getElementById('text-snippet');
-
 const button = document.getElementById('generate-snippet-btn');
+
+// event listener for click events.
 button.addEventListener('click', getSnippet);
 
 function getSnippet() {
-    // call the server using "fetch" to '/generateText' API route
-    // wait for it to return response (read about **promises** and **async await**), take the text that it provided
-    // find the "text-snippet" element and set it's text to the response's text
+    // getting the value of the input.
+    const amountOfSnippetsToGenerate = inputs.value;
+    const requestPayload = {score: amountOfSnippetsToGenerate};
+
+    // speech settings
     let speech = new SpeechSynthesisUtterance();
     speech.lang = "en";
 
-    fetch('/generateText')
-        .then(response => response.text()
-        .then(data => paragraph.innerHTML = data)
-        .then(function(text){
-            speech.text = text;
-            window.speechSynthesis.speak(speech)
-        }
-    ))
-}
\ No newline at end of file
+    // detects number && letters
+    let letters = /^[A-Za-z]+$/;
+    let numbers = /^[0-9]+$/;
+
+    // fetch options.
+    const options = {
+      method: "POST",
+      body: JSON.stringify(requestPayload),
+      headers: { "Content-type": "application/json; charset=UTF-8" }
+    };
+
+    // checks for letters and empty spaces, and bug preventing ..
+      if ( amountOfSnippetsToGenerate === "") {
+        setErrorFor( inputs, 'Number Of Paragraphs , field cannot be empty.')
+    } else if ( amountOfSnippetsToGenerate.match(letters)) {
+        setErrorFor( inputs, 'Number Of Paragraphs , field must contain only numbers.')
+    } else if ( amountOfSnippetsToGenerate > 10 ) {
+        setErrorFor( inputs, 'Number Of Paragraphs cannot be higher than 10.')
+    } else if ( amountOfSnippetsToGenerate <= 0 ) {
+        setErrorFor( inputs, 'Number Of Paragraphs cannot be lower than 1.')
+    } else {
+        fetch( '/api/generateText', options )
+        .then( response => response.text()
+        .then( ( data ) => {
+            if(data.error){
+                alert(data.error)
+            } else if (data === "") {
+                alert('Number Of Paragraphs cannot be symbol!')
```

Comment:
Since you'll improve the validation from previous comment, you wont really need to be this specific here.
However you can change it to "Unknown error has occurred." Just in case